### ISSUE TITLE
Persisting read entries to database

### ISSUE THREAD

**Issue by shakthimaan** [skeeto/elfeed#152]
I am using elpa/elfeed-20151215.938, and use GNU Emacs 24.5.2 (built from source).

Whenever I mark an entry as "read", close Emacs and start it again, I still see the entry in the list. But, when I use "elfeed-db-compact", close Emacs and start it again, the read entries do not show in the list.

Should marking an entry as "read" also write it to the database?

**Comment by skeeto** [comment]
Elfeed should write those changes out automatically when Emacs exits, so
it sounds like your kill-emacs-hook is not being run. This may be due to
an abnormal exiting: a crash or not being given the chance to shut down
gracefully. Once Elfeed has loaded, does that variable in your session
have two Elfeed functions listed?

I'll take a look at that specific version of Emacs. Perhaps there's some
problem with the kill hook.

**Comment by shakthimaan** [comment]
That was a quick reply!

I actually start Emacs in daemon mode and connect to it from an emacsclient. I have bound "C-x w" to start elfeed, and just type "q" to exit elfeed. The client still runs though, until I shutdown the laptop.

**Comment by skeeto** [comment]
I'm betting the Emacs daemon doesn't shut down cleanly with your laptop.
I should add some more save points, perhaps on "q", so that it doesn't
rely only on that hook. If you manually kill the Elfeed buffer it will
save the database. It's harmless since the only state in the buffer
itself is the active filter string.

**Comment by shakthimaan** [comment]
Yes, killing the Elfeed buffer works. Not a bug. Thanks once again for your quick response. Appreciate it!

**Comment by Jesse-Millwood** [comment]
This has been happening to me as well. What is odd is that it worked fine on my other laptop running a different version of emacs (which I also start in daemon mode). I have since had to start using a Windows laptop so I do most of my work in a VM and have noticed this same behaviour now. I usually leave the *elfeed-search* buffer up in the background and refresh it once in a while. Is it better to just manually kill the buffer?

**Comment by groks** [comment]
Try enabling `desktop-save-mode`. `desktop-auto-save-timeout` controls the idle time between desktop saving, which includes flushing the elfeed db.

**Comment by minad** [comment]
Closing, since the issue is old. Saving works as intended (kill-buffer-hook, kill-emacs-hook). If you want to auto save the db, you can also use a timer, see #169. Please ping if you still observe problems.
